pkg/services/grafana-apiserver/service.go

```diff
@@ -140,6 +142,33 @@ func (s *service) start(ctx context.Context) error {
 
 	prepared := server.GenericAPIServer.PrepareRun()
 
+	// TODO: not sure if we can still inject RouteRegister with the new module server setup
```

Comment:
Ideally, we would be able to enable this handler. Its part of our desired APIServer routing that we want to expose through Grafana.

However, `RoutingRegister` is not liking being injected into ModulesServer which is where we `New` the GrafanaAPIServer Service. This is the error I am getting.

```shell
wire: /Users/charandas/go/src/github.com/grafana/grafana/pkg/server/wire.go:430:1: inject InitializeModuleServer: no provider found for github.com/grafana/grafana/pkg/api/routing.RouteRegister
        needed by *github.com/grafana/grafana/pkg/server.ModuleServer in provider set "wireExtsModuleServerSet" (/Users/charandas/go/src/github.com/grafana/grafana/pkg/server/wireexts_oss.go:125:31)
```

For now, I disabled it and left a TODO.

Reply:
@charandas that error means that you need to add the RoutingRegister dependency to the OSS `wireExtsModuleServerSet` (it's already in the enterprise one). Adding that as a dependency may pull in more dependencies, though, and at the moment that means you get to find and fix them _one at a time_ (took me 2 days to get enterprise right ☹️ )

To continue the conversation in slack about wire vs. wireexts, there are a couple of things to know (if I'm explaining too much, I'm sorry, I'm trying to get all my thoughts in one place):

- Our dependencies are initialized with a function (usually `ProvideService`) 
- Those functions may return an interface or a concrete implementation. 
- Returning an interface lets us bind different implementations depending on if we're building Grafana OSS or enterprise

wire.go has the "main" or "general" or maybe "generic" - pick your insufficiently accurate term - wiresets with most of Grafana's dependencies, and the wireexts_* files are for OSS vs. enterprise implementations.

When ProvideService returns a concrete implementation, all you need to add to wire.go is the initializer method name: `thing.ProvideService`

But if `thing.ProvideService` returns an interface, you need to include a `wire.Bind` directive to bind the concrete implementation:

```
thing.ProvideService, // returns a Service interface
wire.Bind(new(thing.Service)), new(thingimplementation.ServiceImplementation) // binds a Service implementation
```

When there's a different implementation for OSS vs. enterprise, those bindings go into the appropriate wireexts_*.go files. Build flags (and some symlinking) control which wireset is built.

Reply:
Mostly got a crash course in wire since joining Grafana and understand it enough, have used it as well. But the separation between various wiresets within Grafana - I could benefit from getting a better understanding of.